Add cron job to disable inactive users

jsf9k opened the pull request:
This is a replacement for cisagov/freeipa-master-tf-module#12.  I think this is a better way, and I will cancel cisagov/freeipa-master-tf-module#12 without merging if this PR is approved.

Thanks to @felddy for pointing out that the domain can be retrieved at run time, which was the key to being able to deploy this cron job via Ansible instead of Terraform.

jsf9k committed:
Add updated Dockerfile.j2 and INSTALL.rst files

These skeleton files come from the latest release of molecule,
generated via the command "molecule init scenario".

jsf9k committed:
Add latest .yamllint generated via molecule init scenario

The generated file contains the truthy: disable bit that was in our
old file.

jsf9k committed:
Add fedora 29 and 30 to molecule configuration

jsf9k committed:
Trigger LGTM

jsf9k committed:
Merge pull request #18 from cisagov/improvement/updated_skeleton_files_from_latest_molecule

Add updated "molecule init scenario" files

jsf9k committed:
Add updates from running pre-commit autoupdate

jsf9k committed:
Merge pull request #17 from cisagov/improvement/updates_from_precommit_autoupdate

Add updates from running pre-commit autoupdate

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_updates

jsf9k committed:
Merge pull request #19 from cisagov/improvement/upstream_updates

Upstream updates

jsf9k committed:
Change "company" name to the latest trend

The new name agrees with the naming guidance that was handed down by
Bobby Thompson.

jsf9k committed:
Merge pull request #20 from cisagov/change_company_name

Change "company" name to the latest trend

jsf9k committed:
Add cron job to disable inactive users, as well as a task to install it

The cron job is a template.

jsf9k committed:
Add prepare.yml playbook to install a cron daemon

This is necessary since we need /etc/cron.daily to exist in the Docker
container.

jsf9k committed:
Add a test to make sure the cron job was installed as expected

felddy commented on templates/disable_inactive_users.j2.sh:
```diff
@@ -0,0 +1,95 @@
+###
+kinit -k -t /etc/krb5.keytab
+
+###
+# Extract the domain from the host keytab
+#
+# This is some potent sed fu.  Match the first line containing
+# "host/something1@something2", then convert something2 to lower case
+# and print it, and finally quit sed.
+###
+domain=$(klist -k /etc/krb5.keytab \
+             | sed --quiet "/.*host\/\(.*\)@\(.*\)/{s//\L\2/p;q}")
```
Nooice. I need some of this sauce over in the OpenVPN script: 
https://github.com/cisagov/ansible-role-openvpn/blob/develop/files/verify-cn.py

TASTY!  🍝

felddy commented on templates/disable_inactive_users.j2.sh:
```diff
@@ -0,0 +1,95 @@
+###
+# Discard the kerberos credentials
+###
+kdestroy
```
You might destroy someone else's creds.  I think it would be better to set the `KRB5CCNAME` environment variable to a private cache file and use that for `kinit` and `kdestroy`.  
e.g.;
```console
root@vpn:~# klist
klist: No credentials cache found (filename: /tmp/krb5cc_0)

root@vpn:~# kinit -k -t /etc/krb5.keytab

root@vpn:~# klist
Ticket cache: FILE:/tmp/krb5cc_0
Default principal: host/vpn.cool.cyber.dhs.gov@COOL.CYBER.DHS.GOV

Valid starting     Expires            Service principal
10/04/19 21:24:10  10/05/19 21:24:10  krbtgt/COOL.CYBER.DHS.GOV@COOL.CYBER.DHS.GOV

root@vpn:~# export KRB5CCNAME=/tmp/foo

root@vpn:~# klist
klist: No credentials cache found (filename: /tmp/foo)

root@vpn:~# kinit -k -t /etc/krb5.keytab

root@vpn:~# klist
Ticket cache: FILE:/tmp/foo
Default principal: host/vpn.cool.cyber.dhs.gov@COOL.CYBER.DHS.GOV

Valid starting     Expires            Service principal
10/04/19 21:25:24  10/05/19 21:25:24  krbtgt/COOL.CYBER.DHS.GOV@COOL.CYBER.DHS.GOV
root@vpn:~# kdestroy 

root@vpn:~# klist
klist: No credentials cache found (filename: /tmp/foo)

root@vpn:~# unset KRB5CCNAME

root@vpn:~# klist
Ticket cache: FILE:/tmp/krb5cc_0
Default principal: host/vpn.cool.cyber.dhs.gov@COOL.CYBER.DHS.GOV

Valid starting     Expires            Service principal
10/04/19 21:24:10  10/05/19 21:24:10  krbtgt/COOL.CYBER.DHS.GOV@COOL.CYBER.DHS.GOV

```

felddy replied:
Something like this could be useful: 
`export KRB5CCNAME=$(mktemp /tmp/krb5cc_disable_active_users.XXXXXXXX)`

jsf9k replied:
This makes a lot of sense and is a better way of handling these creds.  I will make this change.

jsf9k replied:
See 7ea8235.

felddy replied:
Dandy!

felddy reviewed (approved):
Looks great.  I'm stealing some of this.  Please see my comment about `KRB5CCNAME`

jsf9k committed:
Use a temporary kerberos credentials cache file

This way we won't clobber anyone else's.  Thanks again to @felddy for
the suggestion.

jsf9k committed:
Update README to mention the role variable days_before_inactive

jsf9k commented:
This pull request resolves cisagov/freeipa-master-tf-module#13.

felddy reviewed (approved):
Great job.  This will learn those users gud!

![2017-02-04-Three-headed-dog](https://user-images.githubusercontent.com/3229435/66256416-9a671d80-e75b-11e9-9263-9f0eeb3f66ff.gif)

jsf9k committed:
Add default value for days_before_inactive to defaults/main.yml

This is a better place for it than directly in
disable_inactive_users.j2.sh.

jsf9k committed:
Add a test to ensure the days_before_inactive value is correctly applied

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into feature/add_cron_job_to_disable_inactive_users

jsf9k commented:
> Great job. This will learn those users gud!
> 
> ![2017-02-04-Three-headed-dog](https://user-images.githubusercontent.com/3229435/66256416-9a671d80-e75b-11e9-9263-9f0eeb3f66ff.gif)

This is an oddly appropriate cartoon.

dav3r reviewed (approved):
Most excellent work @jsf9k and @felddy - code reviewing at its finest! 🥇
